operator/main.go

```diff
@@ -148,6 +159,79 @@ func runOperator(cmd *cobra.Command) {
 	}
 	close(k8sInitDone)
 
+	k8sversion.Update(k8s.Client(), option.Config)
+	if !k8sversion.Capabilities().MinimalVersionMet {
+		log.Fatalf("Minimal kubernetes version not met: %s < %s",
+			k8sversion.Version(), k8sversion.MinimalVersionConstraint)
+	}
+
+	// Get hostname for identity name of the lease lock holder.
+	// We identify the leader of the operator cluster using hostname.
+	hostID, err := os.Hostname()
```

Comment:
sorry to chime in, I am planning to do something similar at work, so just take this chance to understand more.

Seems like any random id is fine, right ? Any particular reason to have hostName prefix?

Reply:
Just to have some sort of context for the leaseholder ID.